File: angular-frontend/src/app/models/item.ts
import { EntityBase } from "./entity-base";

export class Item extends EntityBase {

  id: string;
  name: string;
  schemaId: string;
  quantity: number;
  properties: ItemProperty[];

  public constructor(init?: Partial<Item>) {
    super();
    Object.assign(this, init);
  }
}

export class ItemProperty {
  key: string;
  value: string;
}

File: dotnet-backend/Models/Item.cs
using System.Collections.Generic;

namespace InventoryManager.Api.Models
{
    public class Item : OwnedEntity
    {
        public string Name { get; set; }
        public string SchemaId { get; set; }
        public int Quantity { get; set; }
        public List<ItemProperty> Properties { get; set; }
    }

    public class ItemProperty
    {
        public string Key { get; set; }
        public string Value { get; set; }
    }
}
